[bakha] 가계부 프로젝트 3주차 1번째 PR

bakhacode opened the pull request:
- 2주차 중반까지도 HTML,CSS에 너무 많은 시간을 보내서 DOM 조작을 제대로 익히지 못했다. JS를 사용해 DOM 조작하는 것도 HTML,CSS와 마찬가지로 거의 처음으로 시도해보는 부분이기도 하고, 심지어 그 둘 보다 훨씬 중요하고 우선순위인 것 같아서 3주차는 온전히 JS에 모든 시간을 쏟는 것을 목표로 잡았다.

bakhacode committed:
Docs: 3주차 간단한 개발계획작성 및 드롭다운 기능구현

2주차에 진행하던 DOM 위주의 학습과 메인페이지에서의 기능구현을 이어서 진행하기로 계획을 설정했다.
제이의 도움으로 중복되는 영역에 이벤트리스너를 달아주는 방법을 배웠다. HTML이 지저분하게 되어있어서 CSS가 좀 어긋나서 수정이 필요하다.

bakhacode committed:
Refactor: info파트 css,js 분리

화면에 보이는 부분은 적은데 생각보다 코드가 길어져서 작업하기 편하게 따로 폴더로 분리해줬다.

bakhacode committed:
Feat: month-year 부분 버블링을 이용한  기능 구현 복습

DOM 조작에 익숙해지기 위해 지피티 참고용으로 구현했던 기능들을 멋지고 든든한 동료 Jay에게 도움을 얻어 구현했다.
결제수단 분류 드롭다운을 구현할 때 사용했던 방식으로 중복되는 코드를 최대한 줄여서 구현했고, 버블링 개념을
다시 한 번 익힐 수 있었다.

namse commented on header/header.js:
```diff
@@ -0,0 +1,76 @@
+// 챗지피티로 확인
+
+function monthIncrease(command) {
+    const num = command === "prev" ? 2 : 0;
+    const [yearNumber, monthNumber, monthName] = getDateEl();
+
+    const newDate = new Date(
+        yearNumber.textContent,
+        monthNumber.textContent - num
+    );
+    setDate([yearNumber, monthNumber, monthName], newDate);
```
코드를 보고 무슨 의도로 뭘 하려는지 모를 수 있을 것 같을 땐
주석을 쓰시거나,
아니면 애초에 그렇게 안짜는 것을 추천합니다.

namse commented on header/header.js:
```diff
@@ -0,0 +1,76 @@
+headerDate.addEventListener("click", (e) => {
+    const $this = e.target;
+    if ($this.tagName === "IMG") {
+        $this.classList.contains("prev-month")
+            ? monthIncrease("prev")
+            : monthIncrease("next");
```
```js
const command = $this.classList.contains("prev-month")
  ? 'prev'
  : 'next';
monthIncrease(command);
```
이래야 했던 것 같기도?

bakhacode committed:
Feat: 금액 인풋바, 결제수단 분류 드롭다운 기능 구현

금액 인풋바에서 숫자를 입력하면 세자리마다 쉼표가 나오게끔 추가해줬다.
결제수단 분류 선택하세요를 클릭하고 나머지 옵션 탭을 누르면 값이 변경되게 구현했다.
